tools/flow9/backends/cpp/cpp.flow

```diff
@@ -160,49 +161,84 @@ bstatement2cpp(backend : Backend<CppInfo>, s : BStatement) -> string {
 				""
 			} else {
 				"return ";
-			}) + bexp2cpp(backend, val) + ";"
+			}) + bexp2cpp(backend, val, type, "") + ";"
 		}
 		BTailCall(fn, args, values, type): {
-			fn  + "(" +superglue(values, rece, ", ") + ");"
+			fn  + "(" +superglue(values, \v -> rece(v, type), ", ") + ");"
 		}
 		BSequenceStatement(statements): {
 			blueprint("
 					{
 						%exps%
 					}",
 				[
-					"exps", superglue(statements, rec, "\n"),
+					"exps", superglue(statements, \statement -> rec(statement, ""), "\n"),
 			]);
 
 		}
-		BSwitchStatement(expr, cases, type): todo();
+		BSwitchStatement(expr, cases, type): {
+			switchType = getBType(expr);
+			caseIndexes = switchType2caseIndexes(switchType, backend);
+			// cases should not be empty. otherwise let there be a crash.
+			// cpp needs a default case in the 'return' case
+			defaultCase = cases[0];
+			blueprint("
+					switch (%value%.index()) {
+						%cases%
+						%default%
+					}",
+				[
+					"value", rece(expr, switchType),
+					"cases", superglue(cases, \case : BCaseStatement -> {
+						blueprint("
+							case %id%: {
+								%body%;
+							}",
+							[
+								"id", eitherMap(lookupTree(caseIndexes, case.id), i2s, ""),
+								"body", rec(case.body, case.id)
+							]);
+					}, ";\n"),
+					"default", "default: " + rec(defaultCase.body, defaultCase.id) + ";",
+			]);
+		}
 	}
 }
 
 
-bexp2cpp(backend : Backend<CppInfo>, bexp : BExp) -> string {
-	rec = \ee -> bexp2cpp(backend, ee);
+bexp2cpp(backend : Backend<CppInfo>, bexp : BExp, resultType : BType, parsedType : string) -> string {
```

Comment:
Again, I'm not sure what parsedType is. Add a comment explaining this a bit better?